infer/SigSuggestion.cc

```diff
@@ -56,16 +56,39 @@ optional<core::AutocorrectSuggestion::Edit> maybeSuggestExtendTSig(core::Context
     return core::AutocorrectSuggestion::Edit{nextLineLoc.value(), fmt::format("{}extend T::Sig\n", prefix)};
 }
 
-core::TypePtr extractArgType(core::Context ctx, cfg::Send &send, core::DispatchComponent &component, int argId) {
+core::TypePtr extractArgType(core::Context ctx, cfg::Send &send, core::DispatchComponent &component,
+                             optional<core::NameRef> keyword, int argId) {
     ENFORCE(component.method.exists());
     const auto &args = component.method.data(ctx)->arguments();
     if (argId >= args.size()) {
         return nullptr;
     }
-    const auto &to = args[argId].type;
+
+    core::TypePtr to = nullptr;
+    if (keyword.has_value()) {
+        auto name = *keyword;
+        for (auto &argInfo : args) {
+            if (argInfo.flags.isKeyword && argInfo.name == name) {
+                to = argInfo.type;
+                break;
+            }
+        }
+        if (to == nullptr) {
+            return nullptr;
+        }
+    } else {
+        auto &argInfo = args[argId];
+        if (argInfo.flags.isKeyword) {
+            return nullptr;
+        }
+
+        to = argInfo.type;
+    }
```

Comment:
Is it worth leaving a comment here somewhere about what might need to be changed to do something smarter for implicit keyword args? (`i.e., to get `T::Hash[Symbol, Integer]`)

Reply:
Yep, definitely. I'll add one.